app/Controllers/CopyController.php

```
@@ -290,6 +290,73 @@ public function updateCopy(Request $request, Response $response, mysqli $db, int
+            if (mb_strlen($numero) > 100) {
+                $numero = mb_substr($numero, 0, 100);
+            }
+            if ($repo->inventoryCodeExists($numero)) {
+                $_SESSION['error_message'] = __('Esiste già una copia con questo numero di inventario.');
+                return $response->withHeader('Location', url("/admin/books/{$bookId}"))->withStatus(302);
+            }
+        } else {
+            $base = !empty($book['numero_inventario']) ? (string) $book['numero_inventario'] : "LIB-{$bookId}";
+            $codes = $repo->allocateInventoryCodes($base, 1);
+            $numero = $codes[0] ?? ($base . '-C1');
+        }
```

PHPMD (2.15.0)

[error] The method createCopy uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)